gremlin-go/driver/graph.go

```diff
@@ -120,3 +129,54 @@ func (p *Path) GetPathObject(key string) (interface{}, error) {
 		return nil, fmt.Errorf("path does not contain a label of '%s'", key)
 	}
 }
+
+// Set describes the necessary methods that need to be implemented to make a user-defined SimpleSet implementation.
+// SimpleSet is the default implementation that should be used but Set can be used for a custom implementation if needed.
+type Set interface {
+	SetToSlice() ([]interface{}, error)
+}
+
+// SimpleSet is a default custom declaration since Go does not natively provide this feature.
+type SimpleSet struct {
+	objects []interface{}
+}
+
+func (s *SimpleSet) SetToSlice() ([]interface{}, error) {
+	return s.objects, nil
+}
+
+func (s *SimpleSet) Add(args ...interface{}) *SimpleSet {
+	for _, val := range args {
+		if !s.Contains(val) {
+			s.objects = append(s.objects, val)
+		}
+	}
+	return s
+}
+
+func (s *SimpleSet) Remove(val interface{}) *SimpleSet {
+	for idx, i := range s.objects {
+		if val == i {
+			// Deletion does not maintain ordering
+			s.objects[idx] = s.objects[len(s.objects)-1]
+			s.objects = s.objects[:len(s.objects)-1]
+			break
+		}
+	}
+	return s
+}
+
+func (s *SimpleSet) Contains(val interface{}) bool {
+	slice, _ := s.SetToSlice()
+	for _, entry := range slice {
+		if val == entry {
+			return true
+		}
+	}
+	return false
+}
+
+// NewSimpleSet constructs a new SimpleSet with filtered values
+func NewSimpleSet(args ...interface{}) *SimpleSet {
+	return new(SimpleSet).Add(args...)
```

Comment:
Instead of returning the `Add` here, just instantiate an empty SimpleSet and invoke `Add` on it in a range loop, then .